hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/scm/ReconNodeManager.java

```diff
@@ -321,4 +321,27 @@ public long getNodeDBKeyCount() throws IOException {
       return nodeCount;
     }
   }
+
+  /**
+   * Remove an existing node from the NodeDB. Explicit removal from admin user.
+   * First this API call removes the node info from NodeManager memory and
+   * if successful, then remove the node finally from NODES table as well.
+   *
+   * @param datanodeDetails Datanode details.
+   */
+  @Override
+  public void removeNode(DatanodeDetails datanodeDetails) throws NodeNotFoundException, IOException {
+    try {
+      nodeDB.delete(datanodeDetails.getUuid());
+    } catch (IOException ioException) {
+      LOG.error("Node {} deletion fails from Node DB.", datanodeDetails.getUuid());
+      // retry to delete the node from node db
+      nodeDB.delete(datanodeDetails.getUuid());
```

Comment:
If failure, do we need retry again? any logic? IMO, we should return failure

Reply:
Ok, will remove retry logic.